documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -28,93 +28,98 @@ We request OpenAPI(Swagger) spec authoring be assigned to engineers who have an
-| M3013	| DeleteMustNotHaveRequestBody | 'Delete' operation must not have a request body. | Error |
-| M2016	| PatchBodyParametersSchemaValidation | Properties of a PATCH request body must not be {0}. PATCH operation: '{1}' Model Definition: '{2}' Property: '{3}' | Error |
-| M2062	| PutResponseResourceValidation | The 200 response model for an ARM PUT operation must have x-ms-azure-resource extension set to true in its hierarchy. Operation: '{0}' Model: '{1}'. | Error |
-| [R3027](#R3027)	| TrackedResourceListByResourceGroup | The tracked resource, '{0}', must have a list by resource group operation. | Error |
-| [R3028](#R3028)	| TrackedResourceListBySubscription | The tracked resource, '{0}', must have a list by subscriptions operation. | Error |
-| [R3011](#R3011)	| DescriptionMustNotBeNodeName | The description provided for a given node (property, parameter, etc.) must not be the same as the name assigned to the node. | Error |
-| [R2020](#R2020) | RequiredPropertiesMissingInResourceModel | A `Resource` model must have `name`, `id` and `type` properties defined. | Error |
+| Id | Rule Name | Output Message |
+| --- | --- | --- |
+| [M3012](#M3012)	| [APIVersionPattern](#M3012)	| API Version must be in the format: yyyy-MM-dd, optionally followed by -preview, -alpha, -beta, -rc, -privatepreview. |
+| M3019	| ARMResourcePropertiesBag	| Top level property names should not be repeated inside the properties bag for ARM resource '{0}'. Properties [{1}] conflict with ARM top level properties. Please rename these. |
+| [M3016](#M3016)	| [BodyPropertiesNamesCamelCase](#M3016) | Property named: "{0}", must follow camelCase style. Example: "{1}". |
+| [M3016](#M3016)	| [DefinitionsPropertiesNamesCamelCase](#M3016)  | Property named: "{0}", for definition: "{1}" must follow camelCase style. Example: "{2}". |
+| M3006	| BodyTopLevelProperties | Top level properties should be one of name, type, id, location, properties, tags, plan, sku, etag, managedBy, identity. Extra properties found: "{0}". |
+| [R3008](#R3008)	| [CollectionObjectPropertiesNaming](#R3008) | Collection object {0} returned by list operation {1} with 'x-ms-pageable' extension, has no property named 'value'. |
+| [R2044](#R2044)	| [InvalidVerbUsed](#R2044) | Permissible values for HTTP Verb are delete,get,put,patch,head,options,post. |
+| [R3023](#R3023)	| [OperationsAPIImplementation](#R3023) | Operations API must be implemented for '{0}'. |
+| [R3007](#R3007)	| [PutGetPatchResponseValidation](#R3007) | {0} has different responses for PUT/GET/PATCH operations. The PUT/GET/PATCH operations must have same schema response. |
+| M3003	| RequiredPropertiesMustExist | Required property does not appear in the list of properties. |
+| M3001 | ResourceModelValidation | Model definition '{0}' must have the properties 'name', 'id' and 'type' in its hierarchy and these properties must be marked as readonly. |
```

Comment:
I believe that once we resolve merge conflicts, we should all rules with `R` and with links and docs. It'd be great i you can verify that, please :)